util/topgen.py

```diff
@@ -85,6 +85,103 @@ def generate_xbars(top, out_path):
         with bind_filepath.open(mode='w', encoding='UTF-8') as fout:
             fout.write(out_bind)
 
+def generate_alert_handler(top, out_path):
+    # default values
+    esc_cnt_dw=32
+    accu_cnt_dw=16
+    lfsr_seed=2**31-1
+    async_on="'0"
+    # leave this constant
+    n_classes=4
+
+    # check if there are any params to be passed through reggen and placed into
+    # the generated package
+    ip_list_in_top = [x["name"].lower() for x in top["module"]]
+    ah_idx = ip_list_in_top.index("alert_handler")
+    if 'localparam' in top['module'][ah_idx]:
+        if 'EscCntDw' in top['module'][ah_idx]['localparam']:
```

Comment:
i can't remember now, but are these localparam's added from the hjson paramlist with the key `local`?

Reply:
yes they are added if you use the localparam keyword in top_earlgrey.hjson.